packages/core/src/render3/node_manipulation.ts

```diff
@@ -804,13 +864,97 @@ function appendProjectedNode(
     if (projectedTNode.type === TNodeType.ElementContainer) {
       let ngContainerChildTNode: TNode|null = projectedTNode.child as TNode;
       while (ngContainerChildTNode) {
-        appendProjectedNode(ngContainerChildTNode, tProjectionNode, currentView, projectionView);
+        appendProjectedNode(
+            ngContainerChildTNode, tProjectionNode, currentView, projectionView, renderParent,
+            anchorNode);
         ngContainerChildTNode = ngContainerChildTNode.next;
       }
     }
 
     if (isLContainer(nodeOrContainer)) {
-      appendChild(nodeOrContainer[NATIVE], tProjectionNode, currentView);
+      renderChild(nodeOrContainer[NATIVE], currentView, anchorNode, renderParent);
+    }
+  }
+}
+
+/**
+ * Adds an {@link LContainer} as a child to an LView in the dynamic case, where the
+ * view may be added at any point in the DOM tree.
+ *
+ * This assures that the linked list of {@link LView} instances is in the same order as they
+ * appear in the DOM, depth first.
+ *
+ * To simply append a child view to a container in the common case, use {@link appendChildView}
+ *
+ * @see appendChildView
+ * @see LView[CHILD_HEAD]
+ * @see LView[CHILD_TAIL]
+ * @see LView[NEXT]
+ *
+ * @param parentLView the LView to add the child container to
+ * @param lContainerToAdd The container to add to the parent
+ */
+export function appendChildViewDynamic(parentLView: LView, lContainerToAdd: LContainer): void {
+  ngDevMode && assertLView(parentLView);
+  ngDevMode && assertLContainer(lContainerToAdd);
+
+  const anchorElement = lContainerToAdd[HOST];
```

Comment:
If you want the host node, you'll need to unwrap it to avoid getting a component view or a styling context.

```suggestion
  const anchorElement = unwrapRNode(lContainerToAdd);
```

Reply:
I've switched to `NATIVE`, because I'm sure that's what I really wanted... however it does concern me that no tests broke from me using `HOST` instead.

cc @mhevery